crates/ide-assists/src/handlers/move_module_to_file.rs

```diff
@@ -107,6 +116,72 @@ mod tests {
 
     use super::*;
 
+    #[test]
+    fn extract_with_specified_path_attr() {
+        check_assist(
+            move_module_to_file,
+            r#"
+//- /main.rs
+#[path="parser/__mod.rs"]
+mod parser;
+//- /parser/__mod.rs
+fn test() {}
+mod $0expr {
+    struct A {}
+}
+"#,
+            r#"
+//- /parser/__mod.rs
+fn test() {}
+mod expr;
+//- /parser/expr.rs
```

Comment:
This won't be correct, the module would need to be at `/parser/__mod/expr.rs` for this to be loaded no?

Reply:
`parser/expr.rs` can be loaded.

![mod_with_path](https://github.com/rust-lang/rust-analyzer/assets/71162630/f4a2b37e-2df5-4ae2-b492-b6229f57aee0)

IMO, with #path specified, `parser/__mod.rs` for mod `parser`(in the following code), can be seen as `parser/mod.rs`
```rust
#[path="parser/__mod.rs"]
mod parser;
```

Reply:
Right, thanks for double checking.